content/en/tracing/guide/code_origins.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Code Origins for Spans
+description: "Learn how to use Code Origins to understand where your spans originate from in your codebase"
+further_reading:
+- link: "/tracing/glossary/"
+  tag: "Documentation"
+  text: "Learn about APM terms and concepts"
+- link: "/tracing/trace_collection/"
+  tag: "Documentation"
+  text: "Learn how to set up APM tracing with your application"
+- link: "/tracing/services/service_page/"
+  tag: "Documentation"
+  text: "Learn more about services in Datadog"
+- link: "/tracing/services/resource_page/"
+  tag: "Documentation"
+  text: "Dive into your resource performance and traces"
+- link: "/dynamic_instrumentation/"
+  tag: "Documentation"
+  text: "Learn how to add custom spans with Dynamic Instrumentation"
+
+---
+
+## Overview
+
+Code Origins captures the exact location in your codebase where APM spans are created. When enabled, it automatically adds file path, line number, and function name to each span, making it easier to:
+
+- Debug performance issues
+- Understand code execution flow
+- Identify performance bottlenecks
+- Onboard new team members
+
+{{< img src="tracing/guide/code_origins/code_origin_details.png" alt="Code Origin Details" style="width:100%;">}}
+
+
+## Getting started
+
+### Prerequisites
+- [Datadog APM][6] configured to capture spans
+- Service meets the [Compatibility Requirements](#compatibility-requirements) below
+- [Source Code Integration][7] enabled (recommended for code previews)
+
+### Compatibility requirements
+
+| Runtime Language | Tracing Library Version | Frameworks |
+|---|---|---|
+| Java | `1.47.0` or later | Spring Boot/Data, gRPC servers, Micronaut 4, Kafka consumers (entry spans only)|
+| Python | `2.15.0` or later | Django, Flask, Starlette and derivatives (entry spans only)|
+| Node.js | `4.49.0` or later | Fastify (supports both entry & exit spans), All other [APM-supported integrations][8] (support exit spans only)|
+
+### Enable Code Origins
+
+Run your service with the below environment variable:
+
+```bash
```

Comment:
I don't know why, but if you make it `shell` instead of `bash`, it will render with a Copy button

```suggestion
```shell
```